### DISCUSSION TITLE
UniTask as a method argument

### DISCUSSION THREAD

**Discussion by gabaguiarp** [Cysharp/UniTask#461] [General]
I am trying to await a UniTask from another script by passing it as an argument to a public method, but if I try to pass a UniTask call similarly to what we do with C# Actions, I get an error message saying: _cannot convert from 'method group' to 'Cysharp.Threading.Tasks.UniTask'_.

I understand what the error means, but I'm also not sure what could be the proper way to pass a UniTask call as an argument.

This is my structure:

**Script A**

        async UniTask UnlockAsync()
        {
            _counterUI.gameObject.SetActive(false);

            await _block.transform.DOScale(Vector3.zero, 0.6f);
            Debug.Log("Block unlocked");

            _block.SetActive(false);
            _isLocked = false;
        }

        void OnGoalRequirementFulfilled()
        {
                _scriptB.RequestCameraFocusAsync(_block.transform, UnlockAsync);
        }

**Script B**

        public async UniTask RequestCameraFocusAsync(Transform focusTarget, UniTask onFocus)
        {
            _focusCameraRig.SetLookAt(focusTarget);
            await SwitchCurrentCameraAsync(_focusCameraRig);
            await onFocus;
            await SwitchToPreviousCameraAsync();
        }

Passing _UnlockAsync_ as the argument in **Script A** causes the aforementioned error.

What am I doing wrong?

**Comment by sebastian-dudzic-spl** [comment]
void OnGoalRequirementFulfilled()
    {
            _scriptB.RequestCameraFocusAsync(_block.transform, UnlockAsync);
    }

change to

 void OnGoalRequirementFulfilled()
    {
            _scriptB.RequestCameraFocusAsync(_block.transform, UnlockAsync());
    }

**Comment by timcassell** [comment]
Just pass a `Func<UniTask>`.
